uni/lib/model/providers/session_provider.dart

```diff
@@ -94,6 +94,6 @@ class SessionProvider extends StateProviderNotifier {
     notifyListeners();
     updateStatus(RequestStatus.failed);
     action?.completeError(RequestStatus.failed);
-    NetworkRouter.onReloginFail();
+    session.persistentSession ? null : NavigationService.logout();
```

Comment:
Question: why checking for persistentSession property?

Reply:
It's from the old logic I think... But I see no reason to keep it

Reply:
I think we need it. Notice that relogin function of session_provider.dart calls [NetworkRouter](https://github.com/NIAEFEUP/project-schrodinger/blob/fix/sigarra-password/uni/lib/controller/networking/network_router.dart).login(...).  If the http response is OK, everything runs according to the expected, otherwise, this will throw an exception causing the reLogin from [session_provider.dart](https://github.com/NIAEFEUP/project-schrodinger/blob/fix/sigarra-password/uni/lib/model/providers/session_provider.dart) to catch the exception. In the end of the catch block, failRelogin(Action? action) is called (as it should, because updateStatus(RequestStatus.failed) and action?.completeError(RequestStatus.failed) are instructions that also make sense in this context). 

This is a problem, for example, when a user has no internet connection and opens the app. As the http request will fail, an exception will be thrown, and he will be immediately logged out because of that catch block in [session_provider.dart](https://github.com/NIAEFEUP/project-schrodinger/blob/fix/sigarra-password/uni/lib/model/providers/session_provider.dart) reLogin(). 

There are other ways to manage this, but I think this check is reasonable in failRelogin because everything there makes sense. We only need to accomplish the case when the session is persistent. 

I actually tested the scenario of the internet connection and that is what happened.

Reply:
Well, if the session isn't persistent, the user will be logged out before they get the chance to need a refresh again. So this reloginfail would never be executed. Am I missing something?

Reply:
Given that the session is persistent, relogin from session_provider.dart is going to try to login:
`_session = await NetworkRouter.login(username, password, faculties, true);`

If you check NetworkRouter.login(...) you will notice that there's the possibility of the session turning persistentSession == false - if i'm not wrong, this is the case when password is changed, for example, which causes the user to get logged out (session is not persistent anymore => logout). This makes sense because before the session was persistent but after changing the password, it's not valid anymore. 

However, if you consider the example I provided before, if the internet connection fails, reLogin from session_provider.dart catches the exception and, in that block, failRelogin(action) is called -> this would cause the user to be logged out if, in failRelogin body, we didn't check if the session was persistent. We don't want this.

I'm not sure if I'm being clear, but it is easier to understand debugging step by step. Anyway, since there was a previous session, reLogin from session_provider.dart will always run. However, if some exception is thrown, we would wrongly log out the user in the function failRelogin(...)... Notice that in this catch block persistentSession is true, but we are logging out the user as well when we call failRelogin, if the check for persistentSession is not present.
```
catch (e) {
      _session = Session(
          studentNumber: username,
          authenticated: false,
          faculties: faculties,
          type: '',
          cookies: '',
          ===> persistentSession: true) <=== ;
      failRelogin(action);
}
```
```
void failRelogin(Completer? action) {
    notifyListeners();
    updateStatus(RequestStatus.failed);
    action?.completeError(RequestStatus.failed);
    ====> NavigationService.logout();  <====
  }
```

Reply:
Yes, that is one of the possible "solutions" if we don't want to check the session in failRelogin. Notice that, before providers, this was the only thing ocurring in the catch:
```
catch (e) {
      final Session renewSession = Session(
          studentNumber: username,
          authenticated: false,
          faculties: faculties,
          type: '',
          cookies: '',
          persistentSession: true);

      action?.completeError(RequestStatus.failed);

      store.dispatch(SaveLoginDataAction(renewSession));
      store.dispatch(SetLoginStatusAction(RequestStatus.failed));
```